datapipelines/src/test/java/io/customer/datapipelines/testutils/core/DataPipelinesTestConfig.kt

```diff
@@ -9,35 +15,61 @@ import io.customer.sdk.CustomerIOBuilder
  *
  * @property cdpApiKey CDP API key to be used for testing
  * @property sdkConfig used to configure the CustomerIO SDK instance for each test separately
- * @property configurePlugins used to configure plugins for analytics instance before
+ * @property analytics used to configure plugins for analytics instance before
  * we add more plugins to it in CustomerIO initialization
  */
 class DataPipelinesTestConfig private constructor(
+    override val arguments: List<TestArgument>,
+    override val diGraph: DIGraphConfiguration,
     val cdpApiKey: String,
-    val sdkConfig: CustomerIOBuilder.() -> Unit,
-    val configurePlugins: Analytics.() -> Unit
-) {
-    class Builder {
+    val sdkConfig: ConfigDSL<CustomerIOBuilder>,
+    val analytics: ConfigDSL<Analytics>
+) : TestConfig {
+    override fun plus(other: TestConfig): DataPipelinesTestConfig {
+        val args = arguments + other.arguments
+        val diGraphConfig = diGraph + other.diGraph
+        if (other !is DataPipelinesTestConfig) {
+            return DataPipelinesTestConfig(
+                arguments = args,
+                diGraph = diGraphConfig,
+                cdpApiKey = cdpApiKey,
+                sdkConfig = sdkConfig,
+                analytics = analytics
+            )
+        }
+
+        return DataPipelinesTestConfig(
+            arguments = args,
+            diGraph = diGraphConfig,
+            cdpApiKey = other.cdpApiKey,
+            sdkConfig = sdkConfig + other.sdkConfig,
+            analytics = analytics + other.analytics
+        )
+    }
```

Comment:
i understand why its like this but can we improve this setup a little?

Reply:
For sure, do ping me if you have any ideas and we can improve this further.